syntax/forth.vim

```diff
@@ -181,36 +180,28 @@ syn keyword forthBlocks BLOCK-INCLUDED BLK
 " numbers
 syn keyword forthMath DECIMAL HEX BASE
 syn match forthInteger '\<-\=[0-9]\+.\=\>'
-syn match forthInteger '\<&-\=[0-9]\+.\=\>'
 syn match forthInteger '\<#-\=[0-9]\+.\=\>'
-" recognize hex and binary numbers, the '$' and '%' notation is for gforth
-syn match forthInteger '\<\$\x*\x\+\>' " *1* --- don't mess
-syn match forthInteger '\<\x*\d\x*\>'  " *2* --- this order!
-syn match forthInteger '\<%[0-1]*[0-1]\+\>'
-syn match forthFloat '\<-\=\d*[.]\=\d\+[DdEe]\d\+\>'
-syn match forthFloat '\<-\=\d*[.]\=\d\+[DdEe][-+]\d\+\>'
+syn match forthInteger '\<\$-\=\x*\x\+\>'
+syn match forthInteger '\<%-\=[0-1]*[0-1]\+\>'
+syn match forthFloat '\<[+-]\=\d*[.]\=\d*[DdEe][+-]\=\d*\>'
```

Comment:
Fix: The forthFloat matching did not cover the examples in the Forth Standard.

Reply:
The first `\d` has to match at least once and the single character collection `[.]` could be replaced with `\.` if it's not an intentional style choice.

```vim
syn match forthFloat '\<[+-]\=\d\+\.\=\d*[DdEe][+-]\=\d*\>'
```